### PR TITLE
Sg/dkg perf

### PR DESCRIPTION
**Pull request by sugh01**

### CODE REVIEW

File: makefile-delete-ns [hunk 1]
```diff
@@ -1,6 +1,8 @@
+include ./envs/charon-dkg-test.env
 # Define the range of indices for generating cluster names
 CLUSTER_INDEX_START := 1
-CLUSTER_INDEX_END := 50
+# CLUSTER_INDEX_END := 10
```
Comment (LukeHackett12) [inline makefile-delete-ns:4]:
```suggestion
```

File: makefile-loadtest-batch-runner [hunk 2]
```diff
@@ -1,11 +1,15 @@
-BATCH_SIZE := 10
+include ./envs/charon-dkg-test.env
+
+BATCH_SIZE := 1
 CLUSTER_INDEX_START := 1
-CLUSTER_INDEX_END := 50
-WAIT_TIME := 10
+# CLUSTER_INDEX_END := FROM ENV
```
Comment (LukeHackett12) [inline makefile-loadtest-batch-runner:5]:
```suggestion
# CLUSTER_INDEX_END := FROM ENV
```

File: .github/workflows/dkg-perf.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: DKG Run
+
+on: 
+    workflow_dispatch:
```
Comment (LukeHackett12) [inline .github/workflows/dkg-perf.yaml:4]: what do we think about adding a repository trigger with a relay url arg? Then could potentially automate tests after a dev relay deploy or things like that

Comment (sugh01) [reply .github/workflows/dkg-perf.yaml:4]: Can do that way, i was thinking to handle the relay url though .env, as we might need to adjust other params too like the cluster size, node nums etc. wdyt?

File: scripts/create-dkg-perf-data.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
+
+#!/bin/bash
+set -uo pipefail
+CLUSTER_NAME_PREFIX="relay-dkg-perf"
+# CLUSTERS=2
+# NODES=3
+# NUM_VALIDATORS=1
+# NETWORK='goerli'
+# CHARON_VERSION='v0.17.0'
```
[the remark this answers is not in the document]
Comment (sugh01) [reply scripts/create-dkg-perf-data.sh]: Definitely!

[the remark this answers is not in the document]
Comment (sugh01) [reply scripts/create-dkg-perf-data.sh]: Cleaned lints

File: .github/workflows/dkg-perf.yaml (hunk 3 above)
Comment (LukeHackett12) [reply .github/workflows/dkg-perf.yaml:4]: I guess that there could be certain things unset in the dkg env file, and then we set them in the workflow instead. Potentially starting with only `CHARON_P2P_RELAYS` could be a good tester.

Comment (sugh01) [reply .github/workflows/dkg-perf.yaml:4]: Yes, we do that way, let me send it in the next push

File: templates/charon-dkg-job-map.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        - name: $NODE_NAME
+          image: ghcr.io/obolnetwork/charon:$CHARON_VERSION
+          imagePullPolicy: Always
+          command:
+            - sh
+            - -ac
+            - >
+              mkdir -p /tmp/.charon;
+              cp /charon/charon-enr-private-key /tmp/.charon/charon-enr-private-key;
+              /usr/local/bin/charon dkg --publish=true --publish-address="https://obol-api-dev.gcp.obol.tech";
```
Comment (LukeHackett12) [inline templates/charon-dkg-job-map.yaml:44]: is it necessary to publish?

**Review by LukeHackett12 — APPROVED** [review]
lgtm! Just one comment

File: templates/charon-dkg-job-map.yaml (hunk 5 above)
Comment (sugh01) [reply templates/charon-dkg-job-map.yaml:44]: Okay will flag this in the next iteration

**Pull request merged**